src/pages/ScreenRouting/mediaScreens/MediaSeries/MediaSeries.tsx

```diff
@@ -1,21 +1,26 @@
 import type { ScreenComponent } from '#types/screens';
-import { getSeriesId } from '#src/utils/media';
+import { getSeriesIdFromCustomParams } from '#src/utils/media';
 import SeriesRedirect from '#src/containers/SeriesRedirect/SeriesRedirect';
 import type { PlaylistItem } from '#types/playlist';
 import Loading from '#src/pages/Loading/Loading';
+import { useWatchHistoryStore } from '#src/stores/WatchHistoryStore';
 
 /**
  * This media screen is used to redirect a series linking media item to an episode page.
  */
-const MediaSeries: ScreenComponent<PlaylistItem> = ({ data, isLoading }) => {
-  const seriesId = getSeriesId(data) || '';
+const MediaSeries: ScreenComponent<PlaylistItem> = ({ data: media, isLoading }) => {
+  const seriesId = getSeriesIdFromCustomParams(media) || '';
 
-  // prevent rendering the SeriesRedirect multiple times when we are loading data
+  // Retrieve watch history for new flow and find an episode of the selected series (if present)
+  const watchHistoryDictionary = useWatchHistoryStore((state) => state.watchHistory);
+  const episodeInProgress = watchHistoryDictionary.find((episode) => episode?.seriesId === media.mediaid);
```

Comment:
It looks like this means that your series watch history changes only work with local storage, not the service, correct?

When we load the watch history, should we check with the series endpoint which series they belong to? Then we don't need to add anything to the service or rely on local storage.

Reply:
Well, there is one issue here.

Let's imagine we have an episode watched. Then we save it in the watch history and reload the Home page. We can simply use `/apps/series?media_ids=xxxx` endpoint to understand whether a media item has a parent series item. But what if one episode is part of two series items? We can't really understand which one to use in this case. 

Maybe we should return the restriction then (one episode <-> one series) and get rid of it only in case we get some complaints.

Reply:
Ya, I think that restriction is pretty reasonable to put back in... honestly it's probably only really an issue for us doing test and debug. Content editors can always clone a media if they need to have it in both places and that will give them better analytics resolution anyhow.